src/net/netsyscall.c

```diff
@@ -1888,6 +1888,43 @@ sysreturn setsockopt(int sockfd,
             goto unimplemented;
         }
         break;
+    case SOL_SOCKET:
+        switch (optname) {
+        case SO_REUSEADDR:
+            if (optlen != sizeof(int))
+                return -EINVAL;
+            if (s->sock.type == SOCK_STREAM) {
+                if (*((int *)optval))
+                    s->info.tcp.lw->so_options |= SOF_REUSEADDR;
+                else
+                    s->info.tcp.lw->so_options &= ~SOF_REUSEADDR;
+            } else {
```

Comment:
I know that stream and dgram are the only options at the moment, but we're likely to at least add SOCK_RAW...maybe just explicitly check for SOCK_DGRAM here (or use switch - case) to mask any unexpected bad behavior in the future? I realize this is probably an issue in other places ... in general it may be safer to not assume any default going forward.

Reply:
Good point, I've updated accordingly.